portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,18 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+// Automatic Slideshow - change image every 3 seconds
+var myIndex = 0;
+carousel();
```

Comment:
It's not great practice to call functions like this, as you have no control over when they get called, it's just whenever the js file gets loaded, which might be before the dom is ready.
A better way to do this is to wait for the event that says the dom is loaded and then call this function.
eg:
document.addEventListener('DOMContentLoaded', (event) => {
    displayCarouselAtIndex(0);
});
https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event

Reply:
fixed - used it in a new carousel but it's not automatic